Libraries/PeriphDrivers/Include/MAX78002/spi.h

```diff
@@ -59,31 +62,72 @@ extern "C" {
 
 /***** Definitions *****/
 
+// clang-format off
+
+// Type or MSMode
+typedef enum {
+    MXC_SPI_TYPE_MASTER = 0,
+    MXC_SPI_TYPE_CONTROLLER = 0,
+    MXC_SPI_TYPE_SLAVE = 1,
+    MXC_SPI_TYPE_TARGET = 1
+} mxc_spi_type_t;
+
+// Target Select (TS) Control Scheme
+typedef enum {
+    MXC_SPI_TSCONTROL_HW_AUTO = 0, // Automatically by hardware
+    MXC_SPI_TSCONTROL_SW_DRV = 1,  // Through software by the driver
+    MXC_SPI_TSCONTROL_SW_APP = 2   // Through software in the application
+} mxc_spi_tscontrol_t;
+
+typedef enum {
+    MXC_SPI_STATE_READY = 0, // Ready for transaction
+    MXC_SPI_STATE_BUSY = 1   // Busy transferring
+} mxc_spi_state_t;
+
+/**
+ * @brief   The list of supported SPI Interface Modes.
+ */
+typedef enum {
+    MXC_SPI_INTERFACE_3WIRE = 0,
+    MXC_SPI_INTERFACE_STANDARD = 1,
+    MXC_SPI_INTERFACE_DUAL = 2,
+    MXC_SPI_INTERFACE_QUAD = 3
+} mxc_spi_interface_t;
+
+/**
+ * @brief The list of SPI clock modes
+ * 
+ * SPI supports four combinations of clock and phase polarity.
+ * 
+ * Clock polarity is controlled using the bit SPIn_CTRL2.cpol 
+ * and determines if the clock is active high or active low
+ * 
+ * Clock phase determines when the data must be stable for sampling
+ */
+typedef enum {
+    MXC_SPI_CLKMODE_0 = 0, // CPOL: 0    CPHA: 0
+    MXC_SPI_CLKMODE_1 = 1, // CPOL: 0    CPHA: 1
+    MXC_SPI_CLKMODE_2 = 2, // CPOL: 1    CPHA: 0
+    MXC_SPI_CLKMODE_3 = 3  // CPOL: 1    CPHA: 1
+} mxc_spi_clkmode_t;
+
+typedef struct {
+    uint32_t index;
+    mxc_gpio_cfg_t pins;
+    uint8_t active_polarity; // Active High (1) or Low (0)
+} mxc_spi_target_t;
+
+///>>> @deprecated
 /**
  * @brief   The list of SPI Widths supported
- *
- * The SPI Width can be set on a per-transaction basis.
- * An example use case of SPI_WIDTH_STANDARD_HALFDUPLEX is
- * given.
- *
- * Using a MAX31865 RTD-to-SPI IC, read back the temperature
- * The IC requires a SPI Read to be executed as
- * 1. Assert SS
- * 2. Write an 8bit register address
- * 3. Read back the 8 bit register
- * 4. Deassert SS
- * This can be accomplished with the STANDARD_HALFDUPLEX width
- * 1. set txData to the address, txLen=1
- * 2. set rxData to a buffer of 1 byte, rxLen=1
- * 3. The driver will transmit the txData, and after completion of
- *    txData begin to recieve data, padding MOSI with DefaultTXData
- *
+ * 
+ * @deprecated.
```

Comment:
Why is `mxc_spi_width_t` deprecated?

Reply:
They do, but it doesn't follow the naming convention in the types (lack of `MXC_` prefix). The `mxc_spi_width_t` is not the best name to use for the interface mode because it can be confused with the message/frame widths. It's also not a common term used to select the interface mode in other peripheral libraries from other vendors (I've only seen ours use it).

This reason is similar to why I added `mxc_spi_clkmode_t` to replace `mxc_spi_mode_t` in SPI v2. The term "mode" has been thrown around a lot and, although it refers to the clock mode, it's been used to represent either "master/slave mode", "clock mode", or the interface mode (dual mode, quad mode, 3-wire mode, and 4-wire mode).

I just want to make the terminology clearer.